ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/Endpoint.java

```diff
@@ -36,13 +35,11 @@
  * used in various Discovery messages.
  */
 public class Endpoint {
-  private final String host;
+  private final Optional<String> host;
   private final int udpPort;
   private final Optional<Integer> tcpPort;
 
-  public Endpoint(final String host, final int udpPort, final Optional<Integer> tcpPort) {
-    checkArgument(
-        host != null && InetAddresses.isInetAddress(host), "host requires a valid IP address");
+  public Endpoint(final Optional<String> host, final int udpPort, final Optional<Integer> tcpPort) {
```

Comment:
non-blocking nitpick, but this signature implies host is optional.  IMO it would be better to expose host as essential, but handle it being nullable internally.  e.g.:
```this.host =  Optional.ofNullable(host)```

$0.02